debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -910,14 +916,36 @@ public synchronized void close() throws SQLException {
                 statementCache.values().forEach(this::cleanupPreparedStatement);
                 statementCache.clear();
                 LOGGER.trace("Closing database connection");
-                conn.close();
+                doClose();
             }
             finally {
                 conn = null;
             }
         }
     }
 
+    private void doClose() throws SQLException {
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        // attempting to close the connection gracefully
+        Future<Object> futureClose = executor.submit(() -> {
+            conn.close();
+            return null;
+        });
+        try {
+            futureClose.get(10, TimeUnit.SECONDS);
+        }
+        catch (ExecutionException e) {
+            throw (SQLException) e.getCause();
+        }
+        catch (TimeoutException | InterruptedException e) {
+            LOGGER.info("Failed to close database connection by calling close(), attempting abort()");
+            conn.abort(Runnable::run);
+        }
+        finally {
+            executor.shutdown();
```

Comment:
Would it be better to use `shutdownNow()`?

Reply:
Agreed